doc/changelog/01-kernel/13563-compact-case-repr.rst

```diff
@@ -0,0 +1,15 @@
+- **Changed:**
+  The term representation of pattern-matchings now uses a compact form that
+  provides a few static guarantees such as eta-expansion of branches and return
+  clauses and is overall more efficient. The most visible user change is that for
+  the time being, the :tacn:`destruct` tactic and its variants generate dummy
+  β-cuts in the branches of the generated proof.
```

Comment:
```suggestion
  The term representation of pattern-matchings now uses a compact form that
  is more efficient.
  This provides additional guarantees such as eta-expansion of branches and return
  clauses. The most visible user change is that
  the :tacn:`destruct` tactic and its variants generate dummy
  β-cuts in the branches of the generated proof.
```

Not sure if "additional" is correct.

Reply:
It's not always more efficient, because converting from and to the expanded representation has a cost. On typical runs it will be more efficient, but YMMV, hence the "overall".

Reply:
OK. "usually more efficient" may be clearer.